src/types.proto

```diff
@@ -2,240 +2,291 @@ syntax = "proto3";
 package types;
 
 // This file is copied from http://github.com/tendermint/abci
+// NOTE: When using custom types, mind the warnings.
+// https://github.com/gogo/protobuf/blob/master/custom_types.md#warnings-and-issues
 
 //----------------------------------------
-// Code types
-
-enum CodeType {
-	OK                    = 0;
-
-	// General response codes, 0 ~ 99
-	InternalError         = 1;
-	EncodingError         = 2;
-	BadNonce              = 3;
-	Unauthorized          = 4;
-	InsufficientFunds     = 5;
-	UnknownRequest        = 6;
-
-	// Reserved for basecoin, 100 ~ 199
-	BaseDuplicateAddress  = 101;
-	BaseEncodingError     = 102;
-	BaseInsufficientFees  = 103;
-	BaseInsufficientFunds = 104;
-	BaseInsufficientGasPrice = 105;
-	BaseInvalidInput      = 106;
-	BaseInvalidOutput     = 107;
-	BaseInvalidPubKey     = 108;
-	BaseInvalidSequence   = 109;
-	BaseInvalidSignature  = 110;
-	BaseUnknownAddress    = 111;
-	BaseUnknownPubKey     = 112;
-	BaseUnknownPlugin     = 113;
-
-	// Reserved for governance, 200 ~ 299
-	GovUnknownEntity      = 201;
-	GovUnknownGroup       = 202;
-	GovUnknownProposal    = 203;
-	GovDuplicateGroup     = 204;
-	GovDuplicateMember    = 205;
-	GovDuplicateProposal  = 206;
-	GovDuplicateVote      = 207;
-	GovInvalidMember      = 208;
-	GovInvalidVote        = 209;
-	GovInvalidVotingPower = 210;
+// Abstract types
 
+// Define these here for compatibility but use tmlibs/common.KVPair.
+message KVPair {
+  bytes key = 1;
+  bytes value = 2;
+}
+
+// Define these here for compatibility but use tmlibs/common.KI64Pair.
+message KI64Pair {
+  bytes key = 1;
+  int64 value = 2;
 }
 
 //----------------------------------------
 // Request types
 
 message Request {
-	oneof value{
-		RequestEcho echo = 1;
-		RequestFlush flush = 2;
-		RequestInfo info = 3;
-		RequestSetOption set_option = 4;
-		RequestDeliverTx deliver_tx = 5;
-		RequestCheckTx check_tx = 6;
-		RequestCommit commit = 7;
-		RequestQuery query = 8;
-		RequestInitChain init_chain = 9;
-		RequestBeginBlock begin_block = 10;
-		RequestEndBlock end_block = 11;
-	}
+  oneof value {
+    RequestEcho echo = 2;
```

Comment:
The `types.proto` file was copy-pasted from the latest version of Tendermint's `types.proto`. I don't know why it's shifted over by one, but I don't want to change it because it could break compatibility.

Reply:
I'm surprised about this. But yeah, makes sense to keep it like this for this PR 👍 
cc @melekes do you have any idea why this was changed to `RequestEcho echo = 2;` in tendermint?

Reply:
No idea